### ISSUE TITLE
Heap overview

### ISSUE THREAD

**Issue by ThatDaleMiller** [teyjus/teyjus#102]
Rob Blanco showed me this bug some weeks ago.  I report it now.

If you run the last goal in the eval.mod file, you get a heap overflow.  The enclosed trace shows that something strange is happening.  Rob claims that this happens "across architectures and compilers".  I was able to reproduce it on my own ubuntu machines running "Teyjus version 2.0-b2".

-Dale

[eval.mod.txt](https://github.com/teyjus/teyjus/files/5998/eval.mod.txt)
[eval.sig.txt](https://github.com/teyjus/teyjus/files/5999/eval.sig.txt)

**Comment by fafounet** [comment]
Looks very similar to issue 53. 
Don't have a text editor on my phone to view the other mod file so I can't say for sure.

**Comment by yvting** [comment]
I have encountered a similar problem. I worked on an implementation of CPS transformation in Danvy&Filinski's style. However, running it on some simple terms resulted in heap overflow. I guess that this bug may be caused by errors in the implementation of the unification algorithm. 

Code is attached.

BTW, the title of this issue seems wrong. Should have been "Heap overflow".

[cps.zip](https://github.com/teyjus/teyjus/files/70348/cps.zip)

**Comment by fafounet** [comment]
Yes it could be a bug in the implementation of the unification algorithm. 
Good luck to find it!
Dale: maybe some hackers attending the reading group on unification at Parsifal could have a look?
